### PR TITLE
feat: storage registry support

### PR DESCRIPTION
**Pull request by CassOnMars**
## Motivation

Adds support for Storage registry contract, plus L2 integration

## Change Summary

Describe the changes being made in 1-2 concise sentences.

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [ ] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [changeset](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#35-adding-changesets)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bugfix, or chore)
- [ ] PR does not require changes to the [protocol](https://github.com/farcasterxyz/protocol)
- [ ] All [commits have been signed](https://github.com/farcasterxyz/hub-monorepo/blob/main/CONTRIBUTING.md#22-signing-commits)

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

---

## PR-Codex overview
This PR focuses on adding storage registry events and related functionality to the codebase. 

### Detailed summary
- Added `RentRegistryEvent` and `StorageAdminRegistryEvent` protobuf messages
- Modified various files to import and use the new protobuf messages
- Added RPC methods for retrieving rent registry events
- Added logging functions for rent registry events
- Added CLI option for L2 RPC URL
- Added methods for submitting rent registry events and storage admin registry events to the mock hub
- Added typeguards for merge rent registry event and merge storage admin registry event hub events
- Modified sync engine constructor to accept L2 events provider

> The following files were skipped due to too many changes: `protobufs/schemas/hub_event.proto`, `apps/hubble/src/rpc/adminServer.ts`, `apps/hubble/src/test/utils.ts`, `packages/core/src/factories.test.ts`, `apps/hubble/src/storage/db/storageRegistryEvent.test.ts`, `packages/core/src/factories.ts`, `apps/hubble/src/storage/engine/index.ts`, `apps/hubble/src/storage/stores/storageEventStore.ts`, `apps/hubble/src/storage/db/storageRegistryEvent.ts`, `apps/hubble/src/eth/l2EventsProvider.test.ts`, `packages/hub-web/src/generated/request_response.ts`, `packages/hub-nodejs/src/generated/request_response.ts`, `packages/core/src/protobufs/generated/request_response.ts`, `packages/hub-nodejs/src/generated/rpc.ts`, `apps/hubble/src/hubble.ts`, `apps/hubble/src/storage/stores/storageEventStore.test.ts`, `packages/hub-web/src/generated/hub_event.ts`, `packages/hub-nodejs/src/generated/hub_event.ts`, `packages/core/src/protobufs/generated/hub_event.ts`, `packages/hub-web/src/generated/storage_event.ts`, `packages/hub-nodejs/src/generated/storage_event.ts`, `packages/core/src/protobufs/generated/storage_event.ts`, `apps/hubble/src/eth/l2EventsProvider.ts`, `packages/hub-web/src/generated/rpc.ts`, `apps/hubble/src/eth/abis.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/hubble/src/rpc/adminServer.ts [hunk 1]
```diff
@@ -191,6 +191,50 @@ export default class AdminServer {
...
+          () => {
+            callback(null, rentRegistryEvent);
+          },
+          (err: HubError) => {
+            callback(toServiceError(err));
+          }
+        );
+      },
+
+      submitStorageAdminRegistryEvent: async (call, callback) => {
```
Comment (sanjayprabhu) [inline apps/hubble/src/rpc/adminServer.ts:217]: Hmm, so why do we need admin endpoints for this? Main concern is these admin submitted messages won't be gossiped out and can cause hubs to get out of sync with each other.

File: apps/hubble/src/cli.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ app
   .command('start')
   .description('Start a Hub')
   .option('-e, --eth-rpc-url <url>', 'RPC URL of a Goerli Ethereum Node')
+  .option('-l, --l2-rpc-url <url>', 'RPC URL of a Goerli Optimism Node')
```
[the remark this answers is not in the document]
Comment (CassOnMars) [reply apps/hubble/src/cli.ts:56]: We're on OP Goerli for now

File: apps/hubble/src/eth/l2EventsProvider.ts [hunk 3]
```diff
@@ -0,0 +1,736 @@
...
+  private _watchBlockNumber: WatchBlockNumber;
+
+  // Whether the historical events have been synced. This is used to avoid syncing the events multiple times.
+  private _isHistoricalSyncDone = false;
+
+  // Number of blocks to wait before processing an event. This is hardcoded to
+  // 6 for now, because that's the threshold beyond which blocks are unlikely
+  // to reorg anymore. 6 blocks represents ~72 seconds on Goerli, so the delay
+  // is not too long.
+  static numConfirmations = 6;
```
[the remark this answers is not in the document]
Comment (CassOnMars) [reply apps/hubble/src/eth/l2EventsProvider.ts:70]: We could reduce it to one if we're following the unsafe (L2 block) / safe (on L1) / finalized (two epochs on L1) paradigm, kept it as is, but if we're good with a single L2 block confirmation then I can adjust it.

File: apps/hubble/src/eth/l2EventsProvider.ts [hunk 4]
```diff
@@ -0,0 +1,736 @@
...
+      // Sync old Rent events
+      await this.syncHistoricalStorageEvents(StorageRegistryEventType.RENT, lastSyncedBlock, toBlock, this._chunkSize);
+
+      // Sync old Storage Admin events
+      await this.syncHistoricalStorageEvents(
+        StorageRegistryEventType.SET_DEPRECATION_TIMESTAMP,
+        lastSyncedBlock,
+        toBlock,
+        this._chunkSize
+      );
```
[the remark this answers is not in the document]
Comment (CassOnMars) [reply apps/hubble/src/eth/l2EventsProvider.ts:436]: The call into `syncHistoricalStorageEvents` is doing iterated chunks of event filters, and there's some events we're not even handling here, and the RPC itself is doing the actual filtering.

File: apps/hubble/src/rpc/adminServer.ts (hunk 1 above)
Comment (CassOnMars) [reply apps/hubble/src/rpc/adminServer.ts:217]: I was adding parity to the existing admin methods for other events, if there's no need for them I can yank 'em.

**Pull request merged**